torchhd/classification.py

```diff
@@ -0,0 +1,101 @@
+import torch
+from tqdm import tqdm
+
+import torchhd
+from torchhd.map import MAP
+
+__all__ = [
+    "EncodingDensityClipped",
+    "classifier_ridge_regression",
+]
+
+
+class EncodingDensityClipped:
```

Comment:
It's a good idea to modularize this encoding. I think this could be a class in the Embeddings module. Could you make it such that it inherits from the nn.Module class? And I would recommend to use `embeddings.Random` instead of the functional form here because then the random vectors are saved when you get the `state_dict` of the model, which is not the case currently. And probably the low and high arguments of the thermometer should also be arguments to the class.

Reply:
`embeddings.Random` is used in the revised version though, I do not fully get the point about `state_dict`.

Reply:
Also moved it to the Embeddings module

Reply:
I'm sorry I explained that a bit hastily. The advantage of using the `Random` class over the functional form is that it attaches the state (i.e., the random vectors) as buffers to the `Module` class. This is something that PyTorch offers and makes sure that when you get the state of your module it will also get the state of all submodules, including the `Random` one, such that you can save them or copy them, or whatever you want to do with it. This ensures that if you train a model on one device and ship the code + state to a different device it will work the same. Whereas if you use the functional form, it would not save the random vectors so when you load the code + state on the new device new random vectors will be created and your model probably won't work very well. Hope this explains it a bit better.

Reply:
Thanks a lot, this explanation makes it clear!